connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -263,36 +266,37 @@ public void startConnect() {
      *
      * @param connName   the name of the connector
      * @param connConfig the intended configuration
-     * @throws IOException          if call to the REST api fails.
+     * @throws ConnectException if the configuration fails to be serialized
      * @throws ConnectRestException if REST api returns error status
      */
-    public void configureConnector(String connName, Map<String, String> connConfig) throws IOException {
+    public String configureConnector(String connName, Map<String, String> connConfig) {
         String url = endpointForResource(String.format("connectors/%s/config", connName));
         ObjectMapper mapper = new ObjectMapper();
-        int status;
+        String content;
         try {
-            String content = mapper.writeValueAsString(connConfig);
-            status = executePut(url, content);
+            content = mapper.writeValueAsString(connConfig);
         } catch (IOException e) {
-            log.error("Could not execute PUT request to " + url, e);
-            throw e;
+            throw new ConnectException("Could not serialize connector configuration and execute PUT request");
         }
-        if (status >= HttpServletResponse.SC_BAD_REQUEST) {
-            throw new ConnectRestException(status, "Could not execute PUT request");
+        Response response = executePut(url, content);
```

Comment:
The `executePut` can actually throw `ConnectException`, so we should reflect that in the JavaDoc

Reply:
Good point. Added